### PR TITLE
STORM-1466: Move the org.apache.thrift7 namespace to something correct/sensible

### PR DESCRIPTION
**Pull request by knusbaum**

### CODE REVIEW

**Comment by ptgoetz** [comment]
+1

**Comment by knusbaum** [comment]
I'm also going to cherry-pick this back into the 1.0 branch.

**Comment by redsanket** [comment]
+1 NB

File: storm-core/pom.xml [hunk 1]
```diff
@@ -618,7 +618,7 @@
                         <relocation>
                             <pattern>org.apache.thrift</pattern>
                             <!-- This pattern is inconsistent for backwards compatibility purposes. -->
-                            <shadedPattern>org.apache.thrift7</shadedPattern>
+                            <shadedPattern>org.apache.storm.thrift</shadedPattern>
```
Comment (d2r) [inline storm-core/pom.xml]: We should remove the old comment about backward compatibility.

Comment (d2r) [inline storm-core/pom.xml]: @ptgoetz no?

**Pull request merged**